index.ts

```diff
@@ -50,85 +51,140 @@ export const setupPlugin: BigQueryPlugin['setupPlugin'] = async (meta) => {
+        throw error
+    }
+}
+
+async function updateBigQueryTableSchema(meta: any, metadata: TableMetadata) {
+        const { global, config } = meta
 
         if (!metadata.schema || !metadata.schema.fields) {
             throw new Error('Can not get metadata for table. Please check if the table schema is defined.')
         }
-
-        const existingFields = metadata.schema.fields
-        const fieldsToAdd = BIG_QUERY_TABLE_FIELDS.filter(
-            ({ name }) => !existingFields.find((f: any) => f.name === name)
-        )
-
-        if (fieldsToAdd.length > 0) {
-            console.info(
-                `Incomplete schema on BigQuery table! Adding the following fields to reach parity: ${JSON.stringify(
-                    fieldsToAdd
-                )}`
+        try {
+            const existingFields = metadata.schema.fields
+            const fieldsToAdd = BIG_QUERY_TABLE_FIELDS.filter(
+                ({ name }) => !existingFields.find((f: any) => f.name === name)
             )
 
-            let result: TableMetadata
-            try {
-                metadata.schema.fields = metadata.schema.fields.concat(fieldsToAdd)
-                ;[result] = await global.bigQueryTable.setMetadata(metadata)
-            } catch (error) {
-                const fieldsToStillAdd = BIG_QUERY_TABLE_FIELDS.filter(
-                    ({ name }) => !result.schema?.fields?.find((f: any) => f.name === name)
+            if (fieldsToAdd.length > 0) {
+                console.info(
+                    `Incomplete schema on BigQuery table! Adding the following fields to reach parity: ${JSON.stringify(
+                        fieldsToAdd
+                    )}`
                 )
 
-                if (fieldsToStillAdd.length > 0) {
-                    throw new Error(
-                        `Tried adding fields ${JSON.stringify(fieldsToAdd)}, but ${JSON.stringify(
-                            fieldsToStillAdd
-                        )} still to add. Can not start plugin.`
+                let result: TableMetadata
+                try {
+                    metadata.schema.fields = metadata.schema.fields.concat(fieldsToAdd)
+                        ;[result] = await global.bigQueryTable.setMetadata(metadata)
+
+                } catch (error) {
+                    console.error("Failed to set Metadata", error)
+
+                    // a different worker already updated the table
+                    const fieldsToStillAdd = BIG_QUERY_TABLE_FIELDS.filter(
+                        ({ name }) => !result.schema?.fields?.find((f: any) => f.name === name)
                     )
+
+                    if (fieldsToStillAdd.length > 0) {
+                        console.error(`Tried adding fields ${JSON.stringify(fieldsToAdd)}, but ${JSON.stringify(
+                            fieldsToStillAdd
+                        )} still to add. Can not start plugin.`)
+                        throw error
```

Comment:
this could also be FetchError, so should be retried

Reply:
jup exactly I changed this as it previously created a custom error here (now just logging that part)